File: LeetCode/Easy/0551-student-attendance-record-i/0551-student-attendance-record-i.go
func checkRecord(s string) bool {
    L := 0
    A := 0
    for i := 0; i < len(s); i++ {
        if s[i] == 'L' {
            L++
            if L > 2 {return false}
        } else if s[i] == 'A' {
            A++
            L = 0
            if A > 1 {return false}
        } else {L = 0}
    }
    return true
}

File: LeetCode/Easy/0551-student-attendance-record-i/0551-student-attendance-record-i.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int A = 0, L = 0;
        for (int i = 0; i < s.length(); ++i){
            if (s[i] == 'L') { if(++L > 2) return false; }
            else if (s[i] == 'A'){
                if(++A > 1) return false;
                L = 0;
            } else L = 0;
        }
        return true;
    }
};